cpp/Platform.Timestamps/Timestamp.h

```diff
@@ -1,25 +1,66 @@
-﻿namespace Platform::Timestamps
-{
-    struct Timestamp : public IEquatable<Timestamp>
-    {
-        public: inline static std::string DefaultFormat = "yyyy.MM.dd hh:mm:ss.fffffff";
-
-        public: std::uint64_t Ticks = 0;
+﻿#pragma once
 
-        public: Timestamp(std::uint64_t ticks) { Ticks = ticks; }
+#include <typeindex>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include "CommonEraClock.h"
 
-        public: Timestamp(DateTime dateTime) : Timestamp((std::uint64_t)dateTime.ToUniversalTime().Ticks) { }
-
-        public: operator DateTime() const { return DateTime((std::int64_t)this->Ticks, DateTimeKind.Utc); }
+namespace Platform::Timestamps
+{
+    class Timestamp
+    {
+        std::uint64_t _ticks;
+    public:
+        static constexpr const char* DefaultFormat = "%Y.%m.%d %H:%M:%S";
+        static constexpr std::uint64_t TicksPerSecond = 10'000'000;
 
-        public: Timestamp(std::uint64_t ticks) : Timestamp(ticks) { }
+        constexpr Timestamp(const std::uint64_t& ticks) noexcept
+            : _ticks(ticks)
+        {
+        }
+        constexpr Timestamp(const std::chrono::time_point<CommonEraClock>& common_time_point)
+            : _ticks(common_time_point.time_since_epoch().count())
+        {
+        }
+        constexpr Timestamp(const Timestamp& timestamp)
+            : _ticks(timestamp.Ticks())
+        {
+        }
+        constexpr Timestamp() = default;
 
-        public: operator ulong() const { return this->Ticks; }
+        constexpr std::uint64_t Ticks() const noexcept
+        {
+            return _ticks;
+        }
+        void Ticks(std::uint64_t ticks)
+        {
+            _ticks = ticks;
+        }
 
-        public: override std::string ToString() { return ((DateTime)this).ToString(DefaultFormat); }
+        auto operator<=>(const Timestamp& other) const noexcept = default;
```

Comment:
In this place `public: operator std::string() const` should be placed. And all other translations should be placed in the same places as the originals.

Reply:
I think it is not really possible, because some methods are changed, and some others were even placed in the std namespace (like hash)

Reply:
It is possible.